### PR TITLE
fix(docs): JSDoc Linting & Fixes

### PR DESCRIPTION
**Pull request by DonovanDMC**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@typescript-eslint/eslint-plugin": "^4.33.0",
     "@typescript-eslint/parser": "^4.33.0",
     "eslint": "^7.32.0",
```
Comment (ArjunSharda) [inline package.json:52]:
```suggestion
    "eslint": "^8.20.0",
```

File: package.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
...
+    "eslint-plugin-jsdoc": "^39.3.3",
     "eslint-plugin-sort-class-members": "^1.14.0",
     "typescript": "^4.4.4"
```
Comment (ArjunSharda) [inline package.json:55]:
```suggestion
    "typescript": "^4.7.4"

File: package.json [hunk 3]
```diff
@@ -50,6 +50,7 @@
     "@typescript-eslint/eslint-plugin": "^4.33.0",
     "@typescript-eslint/parser": "^4.33.0",
```
Comment (ArjunSharda) [inline package.json:51]:
```suggestion
    "@typescript-eslint/parser": "^5.30.7",
```

File: package.json (hunk 1 above)
Comment (DonovanDMC) [inline package.json:52]: bumping eslint is beyond the scope of this pr

File: package.json [hunk 4]
```diff
@@ -50,6 +50,7 @@
     "@typescript-eslint/eslint-plugin": "^4.33.0",
```
Comment (ArjunSharda) [inline package.json:50]:
```suggestion
    "@typescript-eslint/eslint-plugin": "^5.30.7",

**Review by ArjunSharda — CHANGES_REQUESTED** [review]
I've suggested some changes for the package.json file.

File: package.json (hunk 3 above)
Comment (DonovanDMC) [inline package.json:51]: bumping anything is beyond the scope of this pr.

File: package.json (hunk 4 above)
Comment (DonovanDMC) [inline package.json:50]: see previous

Comment (ArjunSharda) [reply package.json:50]: Didn't see before suggesting, sorry.

**Pull request closed without merging**